week2/938_range_sum_bst.py

```diff
@@ -0,0 +1,28 @@
+# https://leetcode.com/problems/range-sum-of-bst/description/
+from typing import Optional
+
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Solution:
+    def range_sum_bst(self, root: Optional[TreeNode], low: int, high: int) -> int:
+        # using nested function
+        def sum_nodes(node):
+            current_node = node
+            current_sum = 0
+            if current_node:
```

Comment:
Try to keep the indentation level low. Human minds do not do a great job when having multiple indentation layer. This code only have one indentation level, but still I would argue the below is much more readable. 
```
            if not current_node:
                return 0
            current_sum = 0
            if low <= current_node.val <= high:
                current_sum += current_node.val
            if current_node.val > low:
                current_sum += sum_nodes(current_node.left)
            if current_node.val < high:
                current_sum += sum_nodes(current_node.right)
            return current_sum
```
In that sense, the following and shorter and `current_sum` variable will be unnecessary. Try to think how to exit functions early if possible. 
```
            if not current_node:
                return 0
            if current_node.val < low:
                return sum_nodes(current_node.right)
            if current_node.val > high:
                return sum_nodes(current_node.left)
            # if low <= current_node.val <= high:
            return sum_nodes(current_node.left) + sum_nodes(current_node.right) + current_node.val
```